### DISCUSSION TITLE
Alternative to App.getInitialProps

### DISCUSSION THREAD

**Discussion by wityan** [vercel/next.js#16018] [Help] [answered]
Is there any good alternative to App.getInitialProps to fetch global data for every page?

**Comment by jamesmosier** [comment, marked as answer]
Currently `getInitialProps` in `_app` is the only way to fetch global data for all pages in one place.

**Comment by mikestopcontinues** [comment]
Imagine a file AppWrapper.js that exports `AppWrapper` and `async getAppWrapperProps(ctx) => {appData}`. You can then import the file into all of your routes like this:

```js
function HomePage(props) => (
  <AppWrapper appData={props.appData}>{props.routeData}</AppWrapper>
);

export async function getStaticProps(ctx) {
  return {props: {...await getAppWrapperProps(ctx), routeData}}
}
```

Unfortunately, you have to do that on every page, but that's the best you've got right now.

**Comment by edsphinx** [comment]
what is the alternative for `getInitialProps` to fetch global data for all pages in one place for app router and not using pages?
